torch/utils/_benchmark/utils/timer.py

```diff
@@ -75,7 +76,27 @@ def repeat(self, repeat=-1, number=-1):
     def autorange(self, callback=None):
         raise NotImplementedError("See `Timer.blocked_autorange.`")
 
-    def blocked_autorange(self, callback=None, min_run_time=0.2):
+    def adaptive_autorange(self, threshold=0.1, max_run_time=10.0, callback=None,
```

Comment:
This replicates a lot of `blocked_autorange`. Is it possible to factor some of the shared logic?

Reply:
Refactored the logic into a common function.